command/agent/config.go

```diff
@@ -92,6 +92,10 @@ type Config struct {
 	// Use normalizedAddrs if you need the host+port to bind to.
 	Addresses *Addresses `hcl:"addresses"`
 
+	// APISocket is used to configure a unix domain socket listener for the
+	// HTTP API
+	APISocket *SocketConfig `hcl:"api_socket"`
```

Comment:
The pedant in me wants to name this `api_unix_socket` because it's the "unix" bit that makes this interesting. All `addresses{}` are all sockets. Using `unix` in the name would match Consul and Vault better as well.

That being said if you don't think anyone wants to type that many characters I will shutup. It is nbd. :sweat_smile: